examples/bri-3/src/bri/identity/bpiSubjectAccounts/agents/bpiSubjectAccountsStorage.agent.ts

```diff
@@ -19,24 +25,18 @@ export class BpiSubjectAccountStorageAgent extends PrismaService {
       throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
     }
 
-    const creatorBpiSubject = new BpiSubject(
-      bpiSubjectAccountModel.CreatorBpiSubject.id,
-      bpiSubjectAccountModel.CreatorBpiSubject.name,
-      bpiSubjectAccountModel.CreatorBpiSubject.description,
-      bpiSubjectAccountModel.CreatorBpiSubject.type,
-      bpiSubjectAccountModel.CreatorBpiSubject.publicKey,
-    );
-    const ownerBpiSubject = new BpiSubject(
-      bpiSubjectAccountModel.OwnerBpiSubject.id,
-      bpiSubjectAccountModel.OwnerBpiSubject.name,
-      bpiSubjectAccountModel.OwnerBpiSubject.description,
-      bpiSubjectAccountModel.OwnerBpiSubject.type,
-      bpiSubjectAccountModel.OwnerBpiSubject.publicKey,
-    );
-    return new BpiSubjectAccount( // TODO: Write generic mapper prismaModel -> domainObject
+    return new BpiSubjectAccount(
       bpiSubjectAccountModel.id,
-      creatorBpiSubject,
-      ownerBpiSubject,
+      this.mapper.map(
+        bpiSubjectAccountModel.CreatorBpiSubject,
+        BpiSubject,
+        BpiSubject,
```

Comment:
twice? do they refer to the BpiSubjectAccount creator and the account owner? @Manik-Jain , if yes then we need to make that clearer

Reply:
@Therecanbeonlyone1969 this is needed because we are mapping from Prisma model to domain model